### ISSUE TITLE
Counts are wrong

### ISSUE THREAD

**Issue by sushilshinde** [topcoder-platform/community-app#3342]
**Steps to reproduce:** 

1. Login with `dan_developer / dantopcoder123` on https://community-app.topcoder-dev.com/challenges
2. Check this SS 
![Topcoder Challenges 2019-09-27 18-16-26](https://user-images.githubusercontent.com/683231/65770203-3210ae80-e153-11e9-8982-6bb375ccd9f5.png)

**Fix required**
Even though metadata is returned correct, counts shown are wrong. Please fix count as exact as metadata (all the counts)

Use these branches for PR :  

1. https://github.com/topcoder-platform/community-app/tree/hot-fix
2. https://github.com/topcoder-platform/topcoder-react-lib/tree/hot-fix

As this is P0, please submit in 12 hours

**Comment by sushilshinde** [comment]
@topcoder-platform/topcodercompetitors @topcoder-platform/tcxcommunity this is open for pickup

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@sushilshinde I investigated perfectly it is working locally , can you please confirm once again?

**Comment by sushilshinde** [comment]
UI doesn't reflect metadata values

**Comment by sushilshinde** [comment]
@topcoder-platform/topcodercompetitors @topcoder-platform/tcxcommunity this is open for pickup

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@sushilshinde replying you in half hour. I am on it

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@sushilshinde on production all working perfectly. I will paste a screenshot for same

**Comment by sushilshinde** [comment]
@topcoder-platform/topcodercompetitors @topcoder-platform/tcxcommunity this is open for pickup

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by codeMinter** [comment]
@sushilshinde Looks like you are looking at earlier response of API calls, can you confirm you see the issue with last response - here is what I see at last response. 

![issue_3342_dev](https://user-images.githubusercontent.com/14284629/65841878-b0807280-e344-11e9-8667-f019ff38a823.png)

**Comment by codeMinter** [comment]
to further test the same, added console logs and see once all challenges are loaded, applying all filters, it gives the right no, so steps to replicate would help here

![issue_3342_logged_in_dev](https://user-images.githubusercontent.com/14284629/65842203-72388280-e347-11e9-85d0-0cd7b339d03f.png)

**Comment by sushilshinde** [comment]
@ajefts seems like this is not re-producible, can you help with the steps?

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by ajefts** [comment]
@sushilshinde @codeMinter Please see this attached video I just made (in production).

https://drive.google.com/file/d/1YnHFE49tzoG016Q1Xtxiir6q8cT1azbs/view

I'm simply refreshing the page a few times.  You can see the challenge counts load with 1 value, then a little later (without doing anything) the counts update again.  Reloading again seems to have yet a different set of counts.

I think some of this is related to the challenges in other groups loading a little later...but I'm not sure if that's all of the issue here.

**Comment by codeMinter** [comment]
@ajefts thanks for the same. Yes, I see the same after several retries on prod, looking into details, I would consider solving it at API level, what is happening is this, we make several calls to backend to fetch next set of 50 challenges, so the values changing experience we have in UI is response we get from these API calls. Sometimes we get exact same metadata while sometimes it changes. So on front-end side we check for metadata in last response and populate it. 

I think if we fix backend to send back right metadata, this shouldn't need change on front-end. Or on front-end, we can show value of 0 and display the final values only when all challenges have loaded. 

So what would be preferred approach?

**Comment by sushilshinde** [comment]
@skyhit can you look at the video that @ajefts shared

https://drive.google.com/file/d/1YnHFE49tzoG016Q1Xtxiir6q8cT1azbs/view

**Comment by sushilshinde** [comment]
@ajefts I think this got also fixed with zurich fix, please have a look on production.

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by ajefts** [comment]
thanks @sushilshinde

**Comment by lakshmiathreya** [comment]
Counts continue to be inconsistent - pls see screenshots attached. @codeMinter @sushilshinde - pls clarify reg nature of fix for this Issue. cc @nithyaasworld 
<img width="1635" alt="Screenshot 2019-10-04 at 11 45 18 AM" src="https://user-images.githubusercontent.com/51733042/66186357-a8efff00-e69f-11e9-809e-fe97f0e2dc04.png">
<img width="1648" alt="Screenshot 2019-10-04 at 11 45 58 AM" src="https://user-images.githubusercontent.com/51733042/66186358-a8efff00-e69f-11e9-8378-c0befae6c65f.png">
<img width="291" alt="Screenshot 2019-10-04 at 11 48 30 AM" src="https://user-images.githubusercontent.com/51733042/66186359-a8efff00-e69f-11e9-84ab-18f1d72b0f74.png">
<img width="1651" alt="Screenshot 2019-10-04 at 11 50 53 AM" src="https://user-images.githubusercontent.com/51733042/66186362-a9889580-e69f-11e9-928e-4a9ddd5a3e70.png">

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by codeMinter** [comment]
@lakshmiathreya can you confirm the counts seen finally are from the last api call metadata ?

**Comment by lakshmiathreya** [comment]
@codeMinter as you can see in the screenshot:
-- the last api call metadata counts match with the response shown in Chrome post login - total count 67.
-- The first api call metadata counts match with the response shown in Chrome prior to login - total count 52. (another Q - should we show Challenges prior to login? Will log a diff issue for this )
-- The counts shown in Firefox dont match counts of first not last api call metadata - total count 60.

**Comment by codeMinter** [comment]
@lakshmiathreya The first 2 scenarios looks right, that is expected behavior
@ajefts @sushilshinde can confirm this too☝️?

Also related to your question - `should we show Challenges prior to login?` - this is expected behavior )

Will check Firefox issue and see what is going wrong there.

**Comment by nithyaasworld** [comment]
@codeMinter Please move this issue back to the card 'QA in PROD in process' if you think we should verify again in Production. Also, add the label 'Ready for Prod QA' back. Thanks.

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by codeMinter** [comment]
> -- The counts shown in Firefox dont match counts of first not last api call metadata - total count 60.

@lakshmiathreya can you retry this? I just tested this on Chrome and Firefox and see same numbers on both. 

![Uploading Firefox.png…]()

<img width="1410" alt="Chrome" src="https://user-images.githubusercontent.com/14284629/66455715-298c7200-ea89-11e9-9dda-914df07dcee9.png">

**Comment by nithyaasworld** [comment]
@codeMinter I am seeing the count is being not correct when followed the steps here https://github.com/topcoder-platform/community-app/issues/3122

Can you check please?

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by codeMinter** [comment]
@nithyaasworld FYI we are waiting on backend fix on this, as metadata is not always returned same by /challenges api call. Once we have that fixed, the numbers should stick around fine. I am putting this on hold till then.

**Comment by crazyk07** [comment]
[500]: Challenge for the updated issue 3342 is creating, rescheduling this event<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by luizrrodrigues** [comment]
@Oanh-and-only-Oanh I think this one is not valid anymore, because this was with v4 API.

And now is related to this one: https://github.com/topcoder-platform/community-app/issues/4488
